src/Shop.java

```diff
@@ -22,57 +22,114 @@ public Shop(String name, int floor, int box, List<String> types){
         this.employees = new LinkedList<Employee>();
     }
 
-    public Shop(String name, int floor, int box){
+    public Shop(String name, int floor, int box) {
         this(name, floor, box, new LinkedList<String>());
     }
 
-    void addProduct(Product product, double qualitity){
+    public void addProduct(Product product, double quantity) {
+        ProductDetails resultProduct = null;
```

Comment:
hey, this part (lines 30-35) looks exactly like `findProductDetails`, let's reuse this method here! :)